python/tskit/trees.py

```diff
@@ -10520,6 +10520,56 @@ def sample_nodes_by_ploidy(self, ploidy):
         sample_node_ids = sample_node_ids.reshape((num_samples_per_individual, ploidy))
         return sample_node_ids
 
+    def map_samples_to_vcf(self, individuals=None):
+        """
+        Returns a list of lists of node IDs, where each sublist contains the
+        sample nodes associated with the same individual.
+        If `individuals` is not specified (the default) nodes for all individuals where
+        all of their nodes are samples are returned. If `individuals` is specified,
+        only the nodes for the specified individuals are returned.
```

Comment:
This looks about right. We also want to return the individual names though, so I'd imagined it would be something like
```
@dataclasses.dataclass
class VcfModelMapping:
    individuals_nodes: np.ndarray
    individuals_name: np.ndarray

def map_to_vcf_model(self, individuals=None, ploidy=None):
     """
     Returns a mapping of the tskit data model to VCF "samples".
...
    return VcfModelMapping(individuals_names, individuals_names)
```